### PR TITLE
exporter/http: gzip request body outside of retryable block

### LINKED ISSUE #1302: Trace exporter request body gzipped on each attempt
The request body is gzipped on [each retry attempt](https://github.com/open-telemetry/opentelemetry-ruby/blob/main/exporter/otlp-http/lib/opentelemetry/exporter/otlp/http/trace_exporter.rb#L139-L144) when exporting traces via otlp-http

### PR DESCRIPTION
**Pull request by tanner-bruce**
Fixes #1302

### CODE REVIEW

File: exporter/otlp-http/lib/opentelemetry/exporter/otlp/http/trace_exporter.rb [hunk 1]
```diff
@@ -134,15 +134,17 @@ def send_bytes(bytes, timeout:) # rubocop:disable Metrics/AbcSize, Metrics/Cyclo
...
-              request.body = if @compression == 'gzip'
-                               request.add_field('Content-Encoding', 'gzip')
-                               Zlib.gzip(bytes)
-                             else
-                               bytes
-                             end
+              request.body = request_body
               request.add_field('Content-Type', 'application/x-protobuf')
+              request.add_field('Content-Encoding', 'gzip') if @compression == 'gzip'
               @headers.each { |key, value| request.add_field(key, value) }
```
Comment (fbogsany) [inline exporter/otlp-http/lib/opentelemetry/exporter/otlp/http/trace_exporter.rb:148]: On retry, s it necessary to redo *any* of the work from the start of the block to here? Can we just hoist all of this out of the `around_request` block?

Comment (plantfansam) [reply exporter/otlp-http/lib/opentelemetry/exporter/otlp/http/trace_exporter.rb:148]: I don't think so....I'd advocate moving it all out of the `around_request` block

**Comment by plantfansam** [comment]
I rolled these changes into https://github.com/open-telemetry/opentelemetry-ruby/pull/1322, hope that's alright

**Comment by tanner-bruce** [comment]
> I rolled these changes into #1322, hope that's alright

Nice, thanks @plantfansam !

**Pull request closed without merging**